packages/interface/src/components/preferences-modal/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * WordPress dependencies
+ */
+import { useViewportMatch } from '@wordpress/compose';
+import {
+	__experimentalNavigatorProvider as NavigatorProvider,
+	__experimentalNavigatorScreen as NavigatorScreen,
+	__experimentalNavigatorButton as NavigatorButton,
+	__experimentalNavigatorBackButton as NavigatorBackButton,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalTruncate as Truncate,
+	FlexItem,
+	Modal,
+	TabPanel,
+	Card,
+	CardHeader,
+	CardBody,
+} from '@wordpress/components';
+import { useMemo, useCallback, useState } from '@wordpress/element';
+import { chevronLeft, chevronRight, Icon } from '@wordpress/icons';
+import { isRTL, __ } from '@wordpress/i18n';
+
+const PREFERENCES_MENU = 'preferences-menu';
+
+export default function PreferencesModal( { closeModal, sections } ) {
```

Comment:
A drawback of having the modal component directly accept the `sections` array for its content is that it makes it impossible to add any freeform content before or after the content (e.g. if there was ever a use case to show an inline notice below the modal header or some kind of footer at the bottom).

An option would be to have another component that acts as the sections:
```jsx
<PreferencesModal>
     // other stuff can be added above or below the sections if needed in the future.
    <PreferencesSections sections={ sections } />
</PreferencesModal>
```

Reply:
True; that would also solve the problem of what happens if there's only one section, as it won't make much sense rendering tabs in that case. `PreferencesSections` could just be responsible for the tabbed interface; maybe best call it `PreferencesTabs` to make that more explicit. (Also because we already have `PreferencesModalSection` that goes inside each tab 😅 )

Reply:
Or `PreferencesModalTabs` to keep to the naming pattern 😄

Reply:
Updated this PR and the testing branch with new component.